### PR TITLE
Remove old Interop.Kernel32.GetVolumeInformation overload

### PR DESCRIPTION
**Pull request by MarcoRossignoli**
closes https://github.com/dotnet/corefx/issues/31867

/cc @JeremyKuhne @pjanotti @krwq

### CODE REVIEW

**Comment by MarcoRossignoli** [comment]
I'm in wip on this...but i've got some question related to this thread https://github.com/dotnet/corefx/pull/31763#discussion_r211360078
Maybe i misread/not fully understood, @krwq recommended to create a wrapper to avoid use of `unsafe` keyword.
We use this api to get 3 different data in 3 different call sites, `DriveFormat` `VolumeLabel` and `fileSystemFlags`.
The other scope of PRs was avoid allocation of unuseful strings. Now if i want consolidate this on one unique helper api i need to return always this 3 data allocating unuseful string.
Other issue is `perhaps even add error handling in there as well` i think i cannot centralize exception handling because on `DriveInfo` there is special treatment to get `friendlier messages for drives`(could throw `DriveNotFoundException`).

File: src/System.IO.FileSystem.DriveInfo/src/System/IO/DriveInfo.Windows.cs [hunk 1]
```diff
@@ -133,24 +121,15 @@ public static DriveInfo[] GetDrives()
...
-                bool success = Interop.Kernel32.SetThreadErrorMode(Interop.Kernel32.SEM_FAILCRITICALERRORS, out oldMode);
-                try
+                using (DisableMediaInsertionPrompt.Create())
                 {
-                    bool r = Interop.Kernel32.GetVolumeInformation(Name, volumeName, volNameLen, out serialNumber, out maxFileNameLen, out fileSystemFlags, fileSystemName, fileSystemNameLen);
-                    if (!r)
+                    if (!Interop.Kernel32.GetVolumeInformation(Name, volumeName, Interop.Kernel32.MAX_PATH + 1, null, null, out int fileSystemFlags, null, 0))
                     {
                         int errorCode = Marshal.GetLastWin32Error();
                         // Win9x appears to return ERROR_INVALID_DATA when a
```
Comment (jkotas) [inline src/System.IO.FileSystem.DriveInfo/src/System/IO/DriveInfo.Windows.cs]: Delete this one-off Win9x specific comment & code while you are on it? The other callsite does not have it.

Comment (MarcoRossignoli) [reply src/System.IO.FileSystem.DriveInfo/src/System/IO/DriveInfo.Windows.cs]: done

**Comment by danmosemsft** [comment]
Thanks @MarcoRossignoli

**Pull request merged**